src/Illuminate/Collections/Arr.php

```diff
@@ -495,6 +495,32 @@ public static function has($array, $keys)
         return true;
     }
 
+    /**
+     * Determine if all keys exist in an array using "dot" notation.
+     *
+     * @param  \ArrayAccess|array  $array
+     * @param  string|array  $keys
+     * @return bool
+     */
+    public static function hasAll($array, $keys)
+    {
+        $keys = (array) $keys;
+
+        if (! $array || $keys === []) {
+            return false;
+        }
+
+        $result = true;
+
+        foreach ($keys as $key) {
+            if (! static::has($array, $key)) {
+                $result = false;
+            }
+        }
+
+        return $result;
```

Comment:
Might be better to bail on first failure.
```suggestion
        foreach ($keys as $key) {
            if (! static::has($array, $key)) {
                return false;
            }
        }

        return true;
```

Reply:
Thanks!